### ISSUE TITLE
[pytorch bert] Is it necessary to have both BertForNextSentencePrediction and BertForSequenceClassification?

### ISSUE THREAD

**Issue by forresti** [NVIDIA/DeepLearningExamples#360]
**Is your feature request related to a problem? Please describe.**
BertForNextSentencePrediction and BertForSequenceClassification appear to be nearly identical, but with the following differences:
- BertForNextSentencePrediction doesn't have dropout, and number of classes is hard-coded to 2.
- BertForSequenceClassification doesn't have dropout, and number of classes is defined by the user.

**Describe the solution you'd like**
Can we delete BertForNextSentencePrediction? I don't actually see it used in the code anywhere.

**Other questions**
Also, do the authors of this codebase about the pros/cons of having dropout before the final classification layer?

**Comment by DaoD** [comment]
The same question. Is there any idea for this?

**Comment by sharathts** [comment]
BertForNextSentencePrediction is for the NSP task only in the pretraining stage. 

You are correct, the only difference is dropout and the number of output classes.

We haven't experimented with dropout/no dropout - it might vary based on the task and the size of the dataset(Do let us know if you obtain better results). Downstream tasks use a dropout layer as seen in the code implemented by BERT authors.

**Comment by sharathts** [comment]
Closing issue. Feel free to reopen if you have more questions.
